backend/src/main/kotlin/code/nebula/cipherquest/service/GameService.kt

```diff
@@ -17,28 +18,29 @@ class GameService(
     @Value("\${application.win-condition}")
     private val winCondition: String,
     private val userLevelService: UserLevelService,
+    private val vectorStoreService: VectorStoreService,
```

Comment:
`GameService` should be responsible just for the game logic, we can remove the `getChatHistory` method and invoke the `VectorStoreService.getMessageHistoryByUserId` method from the `ChatController`.

Reply:
Refactor done!